feugy/tabulous
# feat(web): introduces action visual feedback

### What's in there?

Some actions, especially when performed by peers, deserve some visual feedback: pushing to and popping from a stack, snapping and unsnapping from an anchor, locking and unlocking, and must of all, drawing from and to hand.

Are included here:

- [feat(web): displays visual feedback on push, pop, lock, unlock, snap, unsnap and draw operations](https://github.com/feugy/tabulous/commit/4d5ea2942efa9e3db4237db761dc78f19b7783ec)
- [fix(web): video continues when being the last connected or after leaving game](https://github.com/feugy/tabulous/commit/b2bba8f4aab1a16c97ad5b35e423fd1352ce0e9a)
- [fix(web): indicators are visible while loading UI is displayed](https://github.com/feugy/tabulous/commit/2e8498fca78e95fd37e58d429c77d6b423aaeebf)
- [fix(web): newly invited peer name is undefined when they join](https://github.com/feugy/tabulous/commit/520d5d44ce587b44e710edc55cd3d819ed0a6988)
- [chore(web): adds feedback to indicators 3D manager](https://github.com/feugy/tabulous/commit/27b13d42b9802d39590083c45f02493f75a7158a)
- [chore(web): registers feedback on stack push and pop](https://github.com/feugy/tabulous/commit/e4af7d936fd345a5adfb46372408f4f06b9c72cd)
- [chore(web): registers feedback on anchorable snap and unsnap](https://github.com/feugy/tabulous/commit/3efab92f22ce19a8e61da38fa620eb6fa2aa4e94)
- [chore(web): registers feedback on lockable lock and unlock](https://github.com/feugy/tabulous/commit/3ab51b8eb1c0586f049ae000d35741e7dd14caa0)
- [chore(web): registers feedback when applying peer draws](https://github.com/feugy/tabulous/commit/71ecc876e419d0dfac4b91587705fc999d9a6ca8)
- [chore(web): introduces Feedback component](https://github.com/feugy/tabulous/commit/ad096352fe5e7635d57341c3a00a0314ef3d5eb5)
- [style(web): tries different icon for stack all/pop actions](https://github.com/feugy/tabulous/commit/a72e0ba06a3348cd44c8e20b59e3c798a3d45568)

### How to test?

Create a game that has lockable, drawable, stackable and anchorable meshes, invite other players, and see their actions while they see yours.

Fix for indicators over loading screen:
1. join any game with many stacked meshes 

Fix for video:
1. player A creates a game, and invite player B
2. player B joins
3. player A leaves
   > player B's video stops (it used to continue)

Fix for newly invited player:
1. player A creates a game, and invite player B (do not reload or leave page
5. player B joins, and moves pointer
   > player A should see the player B's pointer with their name (used to contain "undefined")
